include/pipert/ChannelBase.h

```diff
@@ -20,6 +20,9 @@ class ChannelBase {
   int GetPacketSize() const;
   const char* GetName() const;
 
+  int GetFreePacketSize();
```

Comment:
The const question is also valid here. At least for signalling towards the user that nothing bad will happen because of this (even if a mutex is involved).

Reply:
Yes, I agree, naming is not the best here, but first of all I was not sure that it makes sense to use function like that on client side.